src/smt/set_defaults.cpp

```diff
@@ -1024,9 +1027,10 @@ void setDefaults(SmtEngine& smte, LogicInfo& logic)
     // template inference for invariant synthesis, and single invocation
     // techniques.
     bool reqBasicSygus = false;
-    if (options::produceAbducts())
+    if (options::produceAbducts()
+        || options::produceInterpols() != options::ProduceInterpols::NONE)
     {
-      // if doing abduction, we should filter strong solutions
+      // if doing abduction or interpolation, we should filter strong solutions
```

Comment:
Not so sure about that. @ajreynol and @HanielB , is this true for inteprolation?

Reply:
I don't know. I defer to the interpolation users. :)

Reply:
Who are the interpolation users then?

Reply:
maybe, I could add here. But first, what is a strong solution?

Reply:
This refers to logical strength.  For abduction, we always prefer solutions F1 instead of F2 if F2 => F1.

For interpolation, logical strength is not a clear heuristic for which interpolant is best.  I would not recommend enabling this option (you should not change the condition on line 1063).

Reply:
Ideally I think this should be controlled via a separate option so it can be properly evaluated.

Reply:
Yes, which means that we should not hardcode setting this option by default.  (The extra option you talk about is the one that is being forced to true here).

Reply:
Ah, right. :)

Reply:
yeah, I agree that we should not use this option. 

(Also note that the usefulness of a stronger Interpolant in Model Checking is not evident.)